### ISSUE TITLE
Failed to attempt to invoke virtual method

### ISSUE THREAD

**Issue by Dylan901** [TeamNewPipe/NewPipe#7895]
### Checklist

- [X] I am able to reproduce the bug with the [latest version](https://github.com/TeamNewPipe/NewPipe/releases/latest).
- [X] I made sure that there are *no existing issues* - [open](https://github.com/TeamNewPipe/NewPipe/issues) or [closed](https://github.com/TeamNewPipe/NewPipe/issues?q=is%3Aissue+is%3Aclosed) - which I could contribute my information to.
- [X] I have taken the time to fill in all the required details. I understand that the bug report will be dismissed otherwise.
- [X] This issue contains only one bug.
- [X] I have read and understood the [contribution guidelines](https://github.com/TeamNewPipe/NewPipe/blob/dev/.github/CONTRIBUTING.md).

### Affected version

0.21.16

### Steps to reproduce the bug

Version: 0.21.16
Device: Android

Step:
1 Watch any vids

_____
See in logs

### Expected behavior

Will stop saying "Sorry something went wrong" and succesfully attempted to invoke method.

### Actual behavior

Always saying "Sorry something went wrong" and failed to attempt to invoke a method

### Screenshots/Screen recordings

![Screenshot_20220218-225651_NewPipe](https://user-images.githubusercontent.com/95959437/154760633-d90b071c-0f86-4c14-9809-154586a4f301.jpg)
![Screenshot_20220218-225643_NewPipe](https://user-images.githubusercontent.com/95959437/154760638-c3efefce-398d-4e38-b28a-737ef3cea5dc.jpg)

## Exception
* __User Action:__ requested stream
* __Request:__ https://www.youtube.com/watch?v=M7Hl2TwA4O4
* __Content Country:__ GB
* __Content Language:__ en-GB
* __App Language:__ en_GB
* __Service:__ YouTube
* __Version:__ 0.21.15
* __OS:__ Linux samsung/gta4xleea/gta4xl:11/RP1A.200720.012/P615XXU4DUJ1:user/release-keys 11 - 30
<details><summary><b>Crash log </b></summary><p>

```
java.lang.NullPointerException: Attempt to invoke virtual method 'boolean java.lang.String.equals(java.lang.Object)' on a null object reference
	at org.schabi.newpipe.extractor.services.youtube.extractors.YoutubeStreamExtractor.getStreamSegments(YoutubeStreamExtractor.java:1275)
	at org.schabi.newpipe.extractor.stream.StreamInfo.extractOptionalData(StreamInfo.java:333)
	at org.schabi.newpipe.extractor.stream.StreamInfo.getInfo(StreamInfo.java:73)
	at org.schabi.newpipe.extractor.stream.StreamInfo.getInfo(StreamInfo.java:64)
	at org.schabi.newpipe.util.ExtractorHelper.lambda$getStreamInfo$3(ExtractorHelper.java:116)
	at org.schabi.newpipe.util.ExtractorHelper.$r8$lambda$YTHJjScxCJNO1LTCqs3IKy35iyY(Unknown Source:0)
	at org.schabi.newpipe.util.ExtractorHelper$$ExternalSyntheticLambda6.call(Unknown Source:4)
	at io.reactivex.rxjava3.internal.operators.single.SingleFromCallable.subscribeActual(SingleFromCallable.java:43)
	at io.reactivex.rxjava3.core.Single.subscribe(Single.java:4813)
	at io.reactivex.rxjava3.internal.operators.single.SingleDoOnSuccess.subscribeActual(SingleDoOnSuccess.java:35)
	at io.reactivex.rxjava3.core.Single.subscribe(Single.java:4813)
	at io.reactivex.rxjava3.internal.operators.maybe.MaybeFromSingle.subscribeActual(MaybeFromSingle.java:41)
	at io.reactivex.rxjava3.core.Maybe.subscribe(Maybe.java:5330)
	at io.reactivex.rxjava3.internal.operators.maybe.MaybeConcatArray$ConcatMaybeObserver.drain(MaybeConcatArray.java:153)
	at io.reactivex.rxjava3.internal.operators.maybe.MaybeConcatArray$ConcatMaybeObserver.request(MaybeConcatArray.java:78)
	at io.reactivex.rxjava3.internal.operators.flowable.FlowableElementAtMaybe$ElementAtSubscriber.onSubscribe(FlowableElementAtMaybe.java:66)
	at io.reactivex.rxjava3.internal.operators.maybe.MaybeConcatArray.subscribeActual(MaybeConcatArray.java:42)
	at io.reactivex.rxjava3.core.Flowable.subscribe(Flowable.java:15868)
	at io.reactivex.rxjava3.internal.operators.flowable.FlowableElementAtMaybe.subscribeActual(FlowableElementAtMaybe.java:36)
	at io.reactivex.rxjava3.core.Maybe.subscribe(Maybe.java:5330)
	at io.reactivex.rxjava3.internal.operators.maybe.MaybeToSingle.subscribeActual(MaybeToSingle.java:46)
	at io.reactivex.rxjava3.core.Single.subscribe(Single.java:4813)
	at io.reactivex.rxjava3.internal.operators.single.SingleSubscribeOn$SubscribeOnObserver.run(SingleSubscribeOn.java:89)
	at io.reactivex.rxjava3.core.Scheduler$DisposeTask.run(Scheduler.java:644)
	at io.reactivex.rxjava3.internal.schedulers.ScheduledRunnable.run(ScheduledRunnable.java:65)
	at io.reactivex.rxjava3.internal.schedulers.ScheduledRunnable.call(ScheduledRunnable.java:56)
	at java.util.concurrent.FutureTask.run(FutureTask.java:266)
	at java.util.concurrent.ScheduledThreadPoolExecutor$ScheduledFutureTask.run(ScheduledThreadPoolExecutor.java:301)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1167)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:641)
	at java.lang.Thread.run(Thread.java:923)

```
</details>
<hr>

### Affected Android/Custom ROM version

Android 11

### Affected device model

All Affect Samsung Galaxy

**Comment by mhmdanas** [comment]
> 0.21.16

No, you are using 0.21.15, as evidenced by your error report. 0.21.16 doesn't have this issue at all, upgrade to it.

**Comment by Dylan901** [comment]
Duplicate links?

**Comment by mhmdanas** [comment]
It's a duplicate of https://github.com/TeamNewPipe/NewPipe/issues/7792.

Also, my bad, this wasn't fixed in 0.21.16, but it WILL be fixed in the next version, 0.22.0.
